server/src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -427,4 +429,49 @@ public void createTransactionForAccount_tooLargeAmount_throwsException() throws
         IllegalArgumentException.class,
         () -> spannerDao.createTransactionForAccount(accountId, amount, isCredit));
   }
+
+  @Test
+  public void getRecentTransactionsForAccount_valid() throws Exception {
+    ByteArray fromAccountId = UuidConverter.getBytesFromUuid(UUID.randomUUID());
+    ByteArray toAccountId = UuidConverter.getBytesFromUuid(UUID.randomUUID());
+    BigDecimal fromAccountBalance = new BigDecimal(20);
+    BigDecimal toAccountBalance = new BigDecimal(0);
+    BigDecimal amount = new BigDecimal(10);
+    databaseClient.write(
+        ImmutableList.of(
+            Mutation.newInsertBuilder("Account")
+                .set("AccountId")
+                .to(fromAccountId)
+                .set("AccountType")
+                .to(AccountType.UNSPECIFIED_ACCOUNT_TYPE.getNumber())
+                .set("AccountStatus")
+                .to(AccountStatus.UNSPECIFIED_ACCOUNT_STATUS.getNumber())
+                .set("Balance")
+                .to(fromAccountBalance)
+                .set("CreationTimestamp")
+                .to(Value.COMMIT_TIMESTAMP)
+                .build(),
+            Mutation.newInsertBuilder("Account")
+                .set("AccountId")
+                .to(toAccountId)
+                .set("AccountType")
+                .to(AccountType.UNSPECIFIED_ACCOUNT_TYPE.getNumber())
+                .set("AccountStatus")
+                .to(AccountStatus.UNSPECIFIED_ACCOUNT_STATUS.getNumber())
+                .set("Balance")
+                .to(toAccountBalance)
+                .set("CreationTimestamp")
+                .to(Value.COMMIT_TIMESTAMP)
+                .build()));
+    ImmutableMap result = spannerDao.moveAccountBalance(fromAccountId, toAccountId, amount);
+    ImmutableList<TransactionEntry> history =
+        spannerDao.getRecentTransactionsForAccount(
+            fromAccountId, Timestamp.MIN_VALUE, Timestamp.MAX_VALUE);
+    assertThat(history).hasSize(1);
+    assertThat(history.get(0)).isInstanceOf(TransactionEntry.class);
+    assertThat(history.get(0).getAccountId())
+        .isEqualTo(ByteString.copyFrom(fromAccountId.toByteArray()));
+    assertThat(history.get(0).getIsCredit()).isEqualTo(true);
+    assertThat(history.get(0).getAmount()).isEqualTo(amount.toString());
+  }
```

Comment:
Can you add some a test where more than one transaction is involved? To make sure that the function is working for multiple transactions.